### PR TITLE
[10.0] Update wizard_export_fatturapa.py for foreign customers

### PR DESCRIPTION
**Pull request by davidephoto**
Change needed to manage eInvoice to foreign customers, that may have CAP and Proviancia not validating against SDI, and that may have both no VAT# and no FiscalCode.

v. 8.0 #861 

v12: https://github.com/OCA/l10n-italy/pull/886

### CODE REVIEW

**Comment by sergiocorato** [comment]
thanks @davidephoto
check travis

File: l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py [hunk 1]
```diff
@@ -343,7 +343,13 @@ def setCedentePrestatore(self, company, fatturapa):
...
         fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
             DatiAnagrafici = DatiAnagraficiCessionarioType()
-        if not partner.vat and not partner.fiscalcode:
+        if not partner.vat and not partner.fiscalcode and partner.codice_destinatario == 'XXXXXXX':
+            # SDI accpets missing VAT# for foreign customers by setting a fake IdCodice and a valid IdPaese
+            # Otherwise raise error if we havo no VAT# and no Fiscal code
+            fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
+                DatiAnagrafici.IdFiscaleIVA = IdFiscaleType(
+                    IdPaese=partner.country_id.code, IdCodice='99999999999')
+        if not partner.vat and not partner.fiscalcode and not partner.codice_destinatario == 'XXXXXXX':
```
Comment (eLBati) [inline l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py]: I think this check should be also changed in `l10n_it_fatturapa/models/partner.py`: `_check_ftpa_partner_data`

File: l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py [hunk 2]
```diff
@@ -343,7 +343,13 @@ def setCedentePrestatore(self, company, fatturapa):
     def _setDatiAnagraficiCessionario(self, partner, fatturapa):
         fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
             DatiAnagrafici = DatiAnagraficiCessionarioType()
-        if not partner.vat and not partner.fiscalcode:
+        if not partner.vat and not partner.fiscalcode and partner.codice_destinatario == 'XXXXXXX':
```
Comment (eLBati) [inline l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py]: https://github.com/OCA/l10n-italy/pull/861#discussion_r249374170

**Comment by eLBati** [comment]
https://github.com/OCA/l10n-italy/pull/861#pullrequestreview-195001784

**Comment by tafaRU** [comment]
Hi @davidephoto, thanks for your PR!
I'll have you know I'm working on this, and it will be superseeded by a new one in a while.

File: l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py (hunk 2 above)
Comment (davidephoto) [reply l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py]: Se codice_destinatario='XXXXXXX' allora countrycode<>"IT" per definizione, quindi non serve testare il countrycode

File: l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py (hunk 1 above)
Comment (davidephoto) [reply l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py]: You are right, it is at line 81. And we also should not check Provincia and CAP for foreign partners.

File: l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py (hunk 2 above)
Comment (eLBati) [reply l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py]: What if `partner.country_id` is empty?

**Comment by SimoRubi** [comment]
@tafaRU please close this since it has been superseded by your #888

**Pull request closed without merging**